// [[Reason: checker.typeToTypeNode deliberately fails on types that originate from node_modules.]] ////

//// [tests/cases/conformance/node/nodeModulesExportsSourceTs.ts] ////

===================================================================
--- TSC	declarations
+++ DTE	declarations
@@ -1,6 +1,9 @@
 
 
+//// [index.d.ts]
+export declare const a: invalid;
+//# sourceMappingURL=index.d.ts.map
 //// [/.src/node_modules/inner/index.d.ts]
 export { x } from "./other.js";
 //# sourceMappingURL=index.d.ts.map
 //// [/.src/node_modules/inner/other.d.ts]
@@ -11,18 +14,22 @@
 /// [Errors] ////
 
 index.ts(2,23): error TS2307: Cannot find module 'inner/other' or its corresponding type declarations.
 index.ts(3,14): error TS2742: The inferred type of 'a' cannot be named without a reference to './node_modules/inner/other.js'. This is likely not portable. A type annotation is necessary.
+index.ts(3,14): error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations.
 
 
-==== index.ts (2 errors) ====
+==== index.ts (3 errors) ====
     // esm format file
     import { Thing } from "inner/other";
                           ~~~~~~~~~~~~~
 !!! error TS2307: Cannot find module 'inner/other' or its corresponding type declarations.
     export const a = (await import("inner")).x();
                  ~
 !!! error TS2742: The inferred type of 'a' cannot be named without a reference to './node_modules/inner/other.js'. This is likely not portable. A type annotation is necessary.
+                 ~
+!!! error TS9010: Variable must have an explicit type annotation with --isolatedDeclarations.
+!!! related TS9027 index.ts:3:14: Add a type annotation to the variable a.
     import {a as a2} from "package";
 ==== node_modules/inner/index.ts (0 errors) ====
     // esm format file
     export { x } from "./other.js";
